plugin/trino-jmx/src/test/java/io/trino/plugin/jmx/TestJmxSplitManager.java

```diff
@@ -156,6 +157,46 @@ public void testRecordSetProvider()
         }
     }
 
+    @Test
+    public void testNonExistentObjectName()
+            throws Exception
+    {
+        JmxTableHandle jmxTableHandle = metadata.listTables(SESSION, Optional.of(JMX_SCHEMA_NAME)).stream()
+                .map(schemaTableName -> metadata.getTableHandle(SESSION, schemaTableName, Optional.empty(), Optional.empty()))
+                .filter(Objects::nonNull)
+                .filter(tableHandle -> !tableHandle.objectNames().isEmpty())
+                .findFirst()
+                .orElseThrow();
+
+        ImmutableList<String> objectNames = ImmutableList.<String>builder()
+                .addAll(jmxTableHandle.objectNames())
+                .add("JMImplementation:type=Unknown")
+                .build();
+        JmxTableHandle tableHandleWithUnknownObject = new JmxTableHandle(
+                jmxTableHandle.tableName(),
+                objectNames,
+                jmxTableHandle.columnHandles(),
+                jmxTableHandle.liveData(),
+                jmxTableHandle.nodeFilter());
+
+        List<ColumnHandle> columnHandles = ImmutableList.copyOf(metadata.getColumnHandles(SESSION, tableHandleWithUnknownObject).values());
+        ConnectorSplitSource splitSource = splitManager.getSplits(JmxTransactionHandle.INSTANCE, SESSION, tableHandleWithUnknownObject, DynamicFilter.EMPTY, Constraint.alwaysTrue());
+        List<ConnectorSplit> allSplits = getAllSplits(splitSource);
+        assertThat(allSplits).hasSize(nodes.size());
```

Comment:
can we keep assertions together?

move assertThat(allSplits) to the bottom

Reply:
I removed this assertion as it is not relevant for this test.

Reply:
Works for me